cachi2/core/package_managers/yarn/locators.py

```diff
@@ -0,0 +1,236 @@
+from typing import NamedTuple, Protocol
+
+from cachi2.core.errors import PackageRejected
+
+
+class Locator(Protocol):
+    """A generic type for Locator parsers."""
+
+    @property
+    def name(self) -> str:
+        """The name present in this locator.
+
+        Note that the actual package name (defined in its package.json file) might differ from the
+        one present in the locator in case of non-registry dependencies.
+        """
+        ...
+
+    @property
+    def protocol(self) -> str:
+        """The protocol present in this locator."""
+        ...
+
+    @classmethod
+    def from_string(cls, locator: str) -> "Locator":
+        """Parse a Locator from a string."""
+        ...
+
+
+class ExecLocator(NamedTuple):
+    """Locator that handles the exec protocol.
+
+    Sample locator string:
+    dependency-name@exec:./path/to/patch#./path/to/pach::hash=hashstring&locator=workspace%40locator%3A.
+
+    Exec is not supported since it will result in arbitrary code execution.
+    """
+
+    name: str = "n/a"
+    protocol: str = "exec"
+
+    @classmethod
+    def from_string(cls, locator: str) -> "ExecLocator":
+        """Parse a Locator from a string."""
+        raise PackageRejected(
+            reason="The exec protocol is not supported by Cachi2, since it would result in arbitrary code execution.",
+            solution="All dependencies that make use of the exec protocol must be removed from the package.json file.",
+        )
```

Comment:
I was wondering why `resolve_dependency_info` has a check for git but not for exec and then found the check here.

Is this the right place for it?

Reply:
I added a comment to explain my reasoning in the WIP commit. I think the error message should differentiate locators that we can't parse vs the ones we ban.

Not 100% happy of having it here either. Alternatively, we could parse them all and only then ban the ones we forbid in a `_vet_protocols` function.

Reply:
But +1 to having a specific error message for the "known but won't even try to parse" locator types